### PR TITLE
migrate thread-id component to vue3

### PR DESCRIPTION
**Pull request by tahmeedshah**
### General description
_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs
_Link to the related design prototypes (if applicable)_

### Testing instructions
_Provide minimal instructions on how to test this PR._

* Apart from the added feature / bug fix, check overall performance, styling...

## Checklist before merging
* [ ]  Branch name / PR includes the related Jira ticket Id.
* [ ]  Tests to check core implementation / bug fix added.
* [ ]  All checks in Continuous Integration workflow pass.
* [ ]  Feature functionally tested by reviewer(s).
* [ ]  Code reviewed by reviewer(s).
* [ ]  Documentation updated (README, CHANGELOG...) (if required)

### CODE REVIEW

File: app/views/portals/thread-id.vue [hunk 1]
```diff
@@ -1,181 +1,135 @@
...
-              <a class="anchor-margin" name="replies"></a>
-              <span v-if="replies == 0"> {{ $t('linkNoReplies', { count: replies }) }}</span>
-              <span v-if="replies == 1"><i class="fa fa-comment"></i> {{ $t('linkOneReply', { count: replies }) }} </span>
-              <span v-if="replies  > 1"><i class="fa fa-comments"></i> {{ $t('linkXReplies', { count: replies }) }} </span>
-            </em>
+          <em>
+            <a class="anchor-margin" name="replies"></a>
+            <span v-if="replies == 0"> {{ t('linkNoReplies', { count: replies }) }}</span>
+            <span v-if="replies == 1"><i class="fa fa-comment"></i> {{ t('linkOneReply', { count: replies }) }} </span>
+            <span v-if="replies  > 1"><i class="fa fa-comments"></i> {{ t('linkXReplies', { count: replies }) }} </span>
```
Comment (sbilodeau) [inline app/views/portals/thread-id.vue:38]: would have been nice to use plurarization `tc`

File: app/views/portals/thread-id.vue [hunk 2]
```diff
@@ -1,181 +1,135 @@
...
+          </em>
         </template>
       </post>
-
-
-      <div v-if="posts" >
-        <post class="mb-2 border-top" v-for="reply in posts" :key="reply.postId" :post="reply" @refresh="refresh()" />
+      <div v-if="posts">
+        <post class="mb-2 border-top" v-for="reply in posts" :key="reply.postId" :post="reply" @refresh="load" />
```
Comment (sbilodeau) [inline app/views/portals/thread-id.vue:43]: preferable to use event with `()`
```suggestion
        <post class="mb-2 border-top" v-for="reply in posts" :key="reply.postId" :post="reply" @refresh="load()" />
```

**Review by sbilodeau — APPROVED** [review]

File: app/views/portals/thread-id.vue [hunk 3]
```diff
@@ -29,18 +29,18 @@
...
+      <post :post="thread" class="mb-2" @refresh="load()">
         <template v-slot:showReplies="{ replies }">
           <em>
             <a class="anchor-margin" name="replies"></a>
-            <span v-if="replies == 0"> {{ t('linkNoReplies', { count: replies }) }}</span>
-            <span v-if="replies == 1"><i class="fa fa-comment"></i> {{ t('linkOneReply', { count: replies }) }} </span>
-            <span v-if="replies  > 1"><i class="fa fa-comments"></i> {{ t('linkXReplies', { count: replies }) }} </span>
+            <span v-if="replies == 0"> {{ tc('linkNoReplies', { count: replies }) }}</span>
+            <span v-if="replies == 1"><i class="fa fa-comment"></i> {{ tc('linkOneReply', { count: replies }) }} </span>
+            <span v-if="replies  > 1"><i class="fa fa-comments"></i> {{ tc('linkXReplies', { count: replies }) }} </span>
```
Comment (sbilodeau) [inline app/views/portals/thread-id.vue:38]: theorartically it should use only one tranlsationKey no?
eg:
```html
            <span v-if="replies == 0"> {{ tc('linkXReplies', { count: replies }) }}</span>
            <span v-if="replies >= 1"><i class="fa fa-comment"></i> {{ tc('linkXReplies', { count: replies }) }} </span>
```
```json
{
  "linkXReplies": "No replies | One reply |  {count} replies" 
}
```

File: app/app-text/views/portals/forums.json [hunk 4]
```diff
@@ -10,12 +10,13 @@
    "buttonReadReplies": "Read the {count} replies »",
    "linkNoReplies": "No replies",
    "linkOneReply": "One reply",
```
Comment (sbilodeau) [inline app/app-text/views/portals/forums.json:12]: should delete the unused keys no?
```suggestion
```

**Review by sbilodeau — CHANGES_REQUESTED** [review]

**Review by sbilodeau — APPROVED** [review]

File: app/views/portals/thread-id.vue [hunk 5]
```diff
@@ -1,181 +1,135 @@
...
-              <a class="anchor-margin" name="replies"></a>
-              <span v-if="replies == 0"> {{ $t('linkNoReplies', { count: replies }) }}</span>
-              <span v-if="replies == 1"><i class="fa fa-comment"></i> {{ $t('linkOneReply', { count: replies }) }} </span>
-              <span v-if="replies  > 1"><i class="fa fa-comments"></i> {{ $t('linkXReplies', { count: replies }) }} </span>
-            </em>
+          <em>
+            <a class="anchor-margin" name="replies"></a>
+            <span v-if="replies == 0">{{ t('linkXReplies', { count: replies }) }}</span>
+            <span v-if="replies == 1"><i class="fa fa-comment"></i> {{ t('linkXReplies', { count: replies }) }}</span>
+            <span v-if="replies  > 1"><i class="fa fa-comments"></i> {{ t('linkXReplies', { count: replies }) }}</span>
```
Comment (sbilodeau) [inline app/views/portals/thread-id.vue:38]: can we make it on line with `v-if`? I think we can drop the `comment` vs `comments`.
or we keep it with `:class`
```suggestion
            <span>
                <i v-if="replies>0" class="fa" : class="{ 'fa-comment':replies==1, 'fa-comments':replies>1 }"></i>
                {{ t('linkXReplies', { count: replies }) }}
            </span>
```

**Review by sbilodeau — CHANGES_REQUESTED** [review]

**Review by blaisefonseca — APPROVED** [review]

**Review by sbilodeau — APPROVED** [review]

**Pull request merged**